spec/source_file_branch_spec.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require "helper"
+
+if SimpleCov.usable?
+  describe SimpleCov::SourceFile::Branch do
+    let(:results) do
+      [[:if, 0, 1, 4, 10, 4],
+       [:then, 1, 2, 6, 8, 4],
+       [:else, 2, 2, 6, 8, 4],
+       [:then, 1, 1, 6, 9, 5],
+       [:else, 2, 1, 6, 8, 4]]
+    end
+
+    let(:root_branch) do
+      SimpleCov::SourceFile::Branch.new(*(results[0] + [nil]))
+    end
+
+    let(:positive_sub_branch) do
+      SimpleCov::SourceFile::Branch.new(*(results[1] + [0]))
+    end
+
+    let(:negative_sub_branch) do
+      SimpleCov::SourceFile::Branch.new(*(results[2] + [0]))
+    end
+
+    let(:branches) do
+      [root_branch, positive_sub_branch, negative_sub_branch]
+    end
+
+    let(:inline_branches) do
+      [root_branch,
+       SimpleCov::SourceFile::Branch.new(*(results[3] + [0])),
+       SimpleCov::SourceFile::Branch.new(*(results[4] + [0]))]
+    end
+
+    context "A source branch if..else" do
+      it "Is not root branch" do
+        expect(root_branch.root?).to be true
+      end
+
+      it "Is a root branch" do
+        expect(positive_sub_branch.root?).to be       false
+        expect(negative_sub_branch.sub_branch?).to be true
+      end
+
+      it "Has badge + of positive branch" do
+        expect(positive_sub_branch.badge).to eq "+"
+      end
+
+      it "Has badge - of negative branch" do
```

Comment:
this is a bit confusing to read as the tests directly contradict each other (has negative badge, and has positive badge) - I think it might be improved by introducing `describes` that focus on root/positive/negative branch respectively. What do you think?